cloud-media-livestream/keypublisher/main.py

```diff
@@ -0,0 +1,156 @@
+        A string containing the JSON response, status, and headers, to be used
+        by the Flask server.
+    """
+    return (
+        json.dumps({"message": msg, "status": status}),
+        status,
+        {"Content-Type": "application/json"},
+    )
+
+
+def write_secret(secret_id: str, payload: str) -> str:
+    """Writes a secret to Secret Manager.
+
+    If the secret does not exist, it will be created and an initial version
+    added. If the secret does exist, a new version will be created.
+
+    Args:
+        secret_id (string): ID of the secret.
+        payload (string): Secret payload.
+
+    Returns:
+        A string containing the name of the new secret version.
+    """
+    client = secretmanager.SecretManagerServiceClient()
+    project_name = "projects/{}".format(os.environ.get("PROJECT"))
+    try:
+        secret = client.create_secret(
+            request={
+                "parent": project_name,
+                "secret_id": secret_id,
+                "secret": {"replication": {"automatic": {}}},
+            }
+        )
+        secret_name = secret.name
+    except google.api_core.exceptions.AlreadyExists:
+        secret_name = "{}/secrets/{}".format(project_name, secret_id)
+    version = client.add_secret_version(
+        request={"parent": secret_name, "payload": {"data": str.encode(payload)}}
+    )
+    return version.name
+
+
+def validate_environment(cpix_client):
+    required_vars = ["PROJECT"]
+    required_vars.extend(cpix_client.required_env_vars())
+    for required_var in required_vars:
+        if required_var not in os.environ:
+            return http_response(
+                'environment variable "{}" must be set'.format(required_var), 400
+            )
+
+
+@functions_framework.http
+def keys(request):
+    """Fetches encryption keys and uploads key information to Secret Manager.
+
+    Entrypoint of the Cloud Function.
+
+    Args:
+        request (flask.Request): The request object.
+        <https://flask.palletsprojects.com/en/1.1.x/api/#incoming-request-data>
+
+    Returns:
+        The response text, or any set of values that can be turned into a
+        Response object using `make_response`
+        <https://flask.palletsprojects.com/en/1.1.x/api/#flask.make_response>.
+    """
+    if request.method != "POST":
+        return http_response("Only POST requests are supported", 400)
+
+    try:
+        request_json = request.get_json(silent=True)
+        if not request_json:
+            return http_response("no request body was provided", 400)
+
+        media_id = request_json.get("mediaId")
+        if not media_id:
+            return http_response('"mediaId" field must be specified', 400)
+        provider_key = request_json.get("provider")
+        if not provider_key:
+            return http_response(
+                '"provider" field must be specified. supported providers: {}'.format(
+                    ", ".join(PROVIDERS.keys())
+                ),
```

Comment:
fyi: This is fine to use `str.format()` if it goes over multiple lines, or you can still use f-strings.

Reply:
sg, changed to using f-strings